File: Atcoder_#289/ruby/c_coverage.rb
def solve
  n, m = gets.split.map(&:to_i)

  # 各集合を読み込む
  sets = []
  m.times do
    c = gets.to_i
    a = gets.split.map(&:to_i)
    sets << a
  end

  valid_count = 0

  # bit全探索
  (1...2**m).each do |bit|
    covered = Array.new(n+1, false)

    # bitに対応する集合を選ぶ
    m.times do |i|
      if bit[i] == 1
        sets[i].each do |num|
          covered[num] = true
        end
      end
    end

    # 1からNまで全てがカバーされているか確認
    if (1..n).all? { |num| covered[num] }
      valid_count += 1
    end
  end

  # 条件を満たす選び方の数を出力
  puts valid_count
end

solve

File: Atcoder_#289/javascript/c_coverage.js
function solve() {
    const input = require('fs').readFileSync('/dev/stdin', 'utf8').split('\n');
    const [n, m] = input[0].split(' ').map(Number);
  
    // 各集合を読み込む
    let sets = [];
    let index = 1;
    for (let i = 0; i < m; i++) {
      let c = parseInt(input[index]);
      let a = input[index + 1].split(' ').map(Number);
      sets.push(a);
      index += 2;
    }
  
    let validCount = 0;
  
    // bit全探索
    for (let bit = 1; bit < (1 << m); bit++) {
      let covered = Array(n + 1).fill(false);
  
      // bitに対応する集合を選ぶ
      for (let i = 0; i < m; i++) {
        if ((bit >> i) & 1) {
          sets[i].forEach(num => {
            covered[num] = true;
          });
        }
      }
  
      // 1からNまで全てがカバーされているか確認
      if (covered.slice(1).every(Boolean)) {
        validCount++;
      }
    }
  
    // 条件を満たす選び方の数を出力
    console.log(validCount);
  }
  
  solve();
  